presto-main/src/main/java/com/facebook/presto/OutputBuffers.java

```diff
@@ -75,6 +106,13 @@ public boolean isNoMoreBufferIds()
     public void checkValidTransition(OutputBuffers newOutputBuffers)
     {
         requireNonNull(newOutputBuffers, "newOutputBuffers is null");
+        checkArgument(type == newOutputBuffers.getType(), "newOutputBuffers has a different type");
+
+        if (noMoreBufferIds) {
+            checkArgument(this.equals(newOutputBuffers), "Expected buffer to not change after no more buffers is set");
```

Comment:
I think the original order make sense. By moving this check early, it can breaks when `transitions` were received out of order.